src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordObjEquals.cs

```diff
@@ -43,23 +44,27 @@ internal override void GenerateMethodBody(TypeCompilationState compilationState,
 
             try
             {
+                if (_typedRecordEquals.ReturnType.SpecialType != SpecialType.System_Boolean)
+                {
+                    // There is a signature mismatch, an error was reported elsewhere
+                    F.CloseMethod(F.ThrowNull());
+                    return;
+                }
+
                 var paramAccess = F.Parameter(Parameters[0]);
 
                 BoundExpression expression;
                 if (ContainingType.IsStructType())
                 {
-                    throw ExceptionUtilities.Unreachable;
+                    // For record structs:
+                    //      return other is R && Equals((R)other)
```

Comment:
>other is R && Equals((R)other) [](start = 35, length = 30)

The spec suggests a different code - ```other is R temp && Equals(temp)```

Reply:
The underlying IL is probably going to be equivalent though.

---
In reply to: [597087812](https://github.com/dotnet/roslyn/pull/51900#discussion_r597087812) [](ancestors = 597087812)

Reply:
Yes. We don't need to store the temp. I can update the spec if you feel the distinction is meaningful.

---
In reply to: [597090857](https://github.com/dotnet/roslyn/pull/51900#discussion_r597090857) [](ancestors = 597090857,597087812)